src/wp-includes/nav-menu-template.php

```diff
@@ -208,7 +208,8 @@ function wp_nav_menu( $args = array() ) {
 		// Calculate the depth of each menu item with children
 		foreach ( $menu_items_with_children as $menu_item_key => &$menu_item_depth ) {
 			$menu_item_parent = $menu_items_tree[ $menu_item_key ];
-			while ( $menu_item_parent ) {
+			// Do not calculate the depth when the current item has itself as parent.
+			while ( $menu_item_parent && (int) $menu_item_parent !== $menu_item_key ) {
```

Comment:
What happens if an item lists it's own grandchild as a parent, does that cause an infinite loop too? (Same question applies for further descendants)

Reply:
> if an item lists it's own grandchild as a parent, does that cause an infinite loop too

Yea, probably. Since this seems to be coming from plugins thinking it needs a "sensible maximum depth", i.e. the `while()` loop can by stopped after x number of times. For number of sub-menus such max number would be something like 50 perhaps (can you imagine what a menu would look like with 50 levels of sub-menus?).

Reply:
Another thing here: `(int) $menu_item_parent !== $menu_item_key` should probably compare int to int, i.e. typecast both sides. Looking above, there doesn't seem to be a guarantee that `$menu_item_key` is an int, just like `$menu_item_parent`.